test/torchaudio_unittest/functional/numerical_test_impl.py

```diff
@@ -0,0 +1,23 @@
+import torch
+import torchaudio.functional as F
+from torchaudio_unittest import common_utils
+
+
+class NumericalStability(common_utils.TestBaseMixin):
```

Comment:
Couple of comments on this test;
1. I think this fits better in [the existing Lfilter test](https://github.com/pytorch/audio/blob/ea857940de9e3738166989ad3bf1726741a13f04/test/torchaudio_unittest/functional/functional_impl.py#L9). Can you move it there?
2. The intention and the assertion is not straightforward for those without context.
    - Can you add a docstring that explains what this test is supposed to be testing, specifically what `y` should look like if there is no numerical issue?
    - `assert torch.any(y.abs() > 1)` sounds strange. This looks like the ideal implementation of `lfilter` for `float32` has to be larger than `1` at somewhere. While I guess what you want to express is that "when `float32`, it is not guaranteed that `y` meets some criteria, like `y.abs() < 0.8`  (the case of `float64`)

Reply:
> Couple of comments on this test;
> 
> 1. I think this fits better in [the existing Lfilter test](https://github.com/pytorch/audio/blob/ea857940de9e3738166989ad3bf1726741a13f04/test/torchaudio_unittest/functional/functional_impl.py#L9). Can you move it there?

Sure.

> 2. The intention and the assertion is not straightforward for those without context.
>    
>    * Can you add a docstring that explains what this test is supposed to be testing, specifically what `y` should look like if there is no numerical issue?
>    * `assert torch.any(y.abs() > 1)` sounds strange. This looks like the ideal implementation of `lfilter` for `float32` has to be larger than `1` at somewhere. While I guess what you want to express is that "when `float32`, it is not guaranteed that `y` meets some criteria, like `y.abs() < 0.8`  (the case of `float64`)

Yup... I also found it looks kind of wierd and not straightforward. I was thinking about how to get ground truth data without using third party packages.

@mthrok How about pre-compute the ground truth target value offline using SciPy with high order filter, then load it when doing testing? For torch.float32 and torch.float64 we can give tolerance differently, maybe atol=1e-1 or something, then if we solve the numerical issue, we can lower it down. This sounds more like what @cpuhrsch wanted to do.

I'm not sure if it's ok to upload other file formats like `.npy` or `.pth` in test folder, but I saw some `.wav` and `.json` files in the `torchaudio_unittest/assets` folder.

Reply:
This sounds good! Depending on the size of the test assets in my opinion a ```.pth``` file is fine. Of course another alternative is to have a test dependency on scipy, but I'm leaving that up to @mthrok. The advantage there would be further parameterization and ease of use.